pkg/tests/api/alerting/api_notification_channel_test.go

```diff
@@ -0,0 +1,963 @@
+			  "firing": "Labels:\n - alertname = PagerdutyAlert\nAnnotations:\nSource: \n",
+			  "num_firing": "1",
+			  "num_resolved": "0",
+			  "resolved": ""
+			}
+		  },
+		  "client": "Grafana",
+		  "client_url": "http://localhost:3000/",
+		  "links": [
+			{
+			  "href": "http://localhost:3000/",
+			  "text": "External URL"
+			}
+		  ]
+		}`,
+	},
+	"dingding_recv/dingding_test": {
+		`{
+		  "link": {
+			"messageUrl": "dingtalk://dingtalkclient/page/link?pc_slide=false&url=http%3A%2Flocalhost%3A3000%2Falerting%2Flist",
+			"text": "\n**Firing**\nLabels:\n - alertname = DingDingAlert\nAnnotations:\nSource: \n\n\n\n\n",
+			"title": "[firing:1]  (DingDingAlert)"
+		  },
+		  "msgtype": "link"
+		}`,
+	},
+	"teams_recv/teams_test": {
+		`{
+		  "@context": "http://schema.org/extensions",
+		  "@type": "MessageCard",
+		  "potentialAction": [
+			{
+			  "@context": "http://schema.org",
+			  "@type": "OpenUri",
+			  "name": "View Rule",
+			  "targets": [
+				{
+				  "os": "default",
+				  "uri": "http:/localhost:3000/alerting/list"
+				}
+			  ]
+			}
+		  ],
+		  "sections": [
+			{
+			  "text": "\n**Firing**\nLabels:\n - alertname = TeamsAlert\nAnnotations:\nSource: \n\n\n\n\n",
+			  "title": "Details"
+			}
+		  ],
+		  "summary": "[firing:1]  (TeamsAlert)",
+		  "themeColor": "#D63232",
+		  "title": "[firing:1]  (TeamsAlert)"
+		}`,
+	},
+	"webhook_recv/webhook_test": {
+		`{
+		  "receiver": "webhook_recv",
+		  "status": "firing",
+		  "alerts": [
+			{
+			  "status": "firing",
+			  "labels": {
+				"alertname": "WebhookAlert"
+			  },
+			  "annotations": {},
+			  "startsAt": "%s",
+			  "endsAt": "0001-01-01T00:00:00Z",
+			  "generatorURL": "",
+			  "fingerprint": "a0e2a99fc56fce63"
+			}
+		  ],
+		  "groupLabels": {},
+		  "commonLabels": {
+			"alertname": "WebhookAlert"
+		  },
+		  "commonAnnotations": {},
+		  "externalURL": "http://localhost:3000/",
+		  "version": "1",
+		  "groupKey": "{}/{alertname=\"WebhookAlert\"}:{}",
+		  "truncatedAlerts": 0,
+		  "title": "[FIRING:1]  (WebhookAlert)",
+		  "state": "alerting",
+		  "message": "\n**Firing**\nLabels:\n - alertname = WebhookAlert\nAnnotations:\nSource: \n\n\n\n\n"
+		}`,
+	},
+}
+
+func getAlertmanagerConfig(channelAddr string) string {
+	return strings.ReplaceAll(alertmanagerConfig, "CHANNEL_ADDR", channelAddr)
+}
+
+func getExpAlertmanagerConfigFromAPI(channelAddr string) string {
+	return strings.ReplaceAll(expAlertmanagerConfigFromAPI, "CHANNEL_ADDR", channelAddr)
+}
+
+// alertNames are name of alerts to be sent. This should be in sync with
+// the routes that we define in Alertmanager config.
+// EmailAlert and TelegramAlert are missing because they don't
+// send a JSON. Email and POST body are yet to be supported in the tests.
+var alertNames = []string{"DingDingAlert", "SlackAlert1", "SlackAlert2", "PagerdutyAlert", "TeamsAlert", "WebhookAlert"}
+
+func getRulesConfig(t *testing.T) string {
+	interval, err := model.ParseDuration("10s")
```

Comment:
```suggestion
        t.Helper()
	interval, err := model.ParseDuration("10s")
```

nit: Can we move all this to the bottom? Test should be at the top of the file while helpers / fixtures at the bottom.